homeassistant/components/ping/binary_sensor.py

```diff
@@ -131,20 +132,28 @@ def __init__(self, hass, host, count) -> None:
 class PingDataICMPLib(PingData):
     """The Class for handling the data retrieval using icmplib."""
 
-    def ping(self):
-        """Send ICMP echo request and return details."""
-        return icmp_ping(self._ip_address, count=self._count)
-
     async def async_update(self) -> None:
         """Retrieve the latest details from the host."""
-        data = await self.hass.async_add_executor_job(self.ping)
+        _LOGGER.warning("ping address: %s", self._ip_address)
+        data = await self.hass.async_add_executor_job(
+            partial(
+                icmp_ping,
+                self._ip_address,
+                count=self._count,
+                id=async_get_next_ping_id(self.hass),
```

Comment:
You're passing a method marked for async to be executed in the threadpool. Probably just make the method sync?

Reply:
Oh snap nevermind, we're calling the function here 👍